lua/entities/base_wire_entity.lua

```diff
@@ -292,25 +310,31 @@ function ENT:SetOverlayText( txt )
 	if not self.OverlayData then
 		self.OverlayData = {}
 	end
-
-	if txt and #txt > 12000 then
-		txt = string.sub(txt,1,12000) -- I have tested this and 12000 chars is enough to cover the entire screen at 1920x1080. You're unlikely to need more
+	if txt and #txt > 512 then
+		txt = string.sub(txt,1,512) -- I have tested this and 12000 chars is enough to cover the entire screen at 1920x1080. You're unlikely to need more
 	end
-
 	self.OverlayData.txt = txt
+	if CLIENT then return end
+	if not self.playersRequestingOverlayNumeric or #self.playersRequestingOverlayNumeric == 0 then return end
 
-	if not self.OverlayData_UpdateTime then	self.OverlayData_UpdateTime = {} end
-	self.OverlayData_UpdateTime.time = CurTime()
+	net.Start( "wire_overlay_txt", true )
+		net.WriteEntity( self )
+		net.WriteString( self.OverlayData.txt )
+	net.Send(self.playersRequestingOverlayNumeric)
```

Comment:
Invalid players will need to be removed from the table. This should also prevent leaking.
```lua
local i = 1
while i <= #self.playersRequestingOverlayNumeric do
   if self.playersRequestingOverlayNumeric[i]:IsValid() then
      i = i + 1
   else
      table.remove(self.playersRequestingOverlayNumeric, i)
   end
end
```

Reply:
Forgot to mention, this would occur if a player disconnects while looking at a wire entity.

Reply:
Or you can do this and make the table player indexed instead.

```lua
local plyToSend = {}
for ply in pairs(self.playersRequestingOverlayNumeric) do
   if ply:IsValid() then plyToSend[#plyToSend+1] = ply else self.playersRequestingOverlayNumeric[ply] = nil end
end
```

Reply:
id rather not make it player indexed as id have to use the vastly slower pairs instead of ipairs in a function that may be called every tick if the data is being updated at the rate.

Reply:
The best place to add a function to scan the table for invalid players is a `PlayerDisconnected` hook.  Id rather not add expensive looping checks for validity inside the function that may be called once per tick.